extensions/applicationinsights-properties-js/src/Context/Internal.ts

```diff
@@ -37,11 +37,13 @@ export class Internal implements IInternal {
     /**
      * Constructs a new instance of the internal telemetry data class.
      */
-    constructor(config: IPropertiesConfig) {
+    constructor(config: IPropertiesConfig, unloadHookContainer?: IUnloadHookContainer) {
         
-        onConfigChange((config), () => {
+        let unloadHook = onConfigChange((config), () => {
             let prefix =  config.sdkExtension;
             this.sdkVersion = (prefix ? prefix + "_" : "") + "javascript:" + Version;
         });
+
+        unloadHookContainer && unloadHookContainer.add(unloadHook);
```

Comment:
We always need to "register" (`add()`) the returned value from the `onConfigChange()` because it has internal side-effects that indirectly cause the callback function (the listener) to be associated with and referenced configuration item. And if the `unloadHook` is never called all of these internal references would cause this class instance to never be garbage collected, which would result in a "memory leak" (because of the possibility of multiple instances getting left behind as referenced -- lots of assumptions in this statement and this edge case)